### PR TITLE
Fix initial algolia import + refactor the import component

### PR DESCRIPTION
**Pull request by andrzejewsky**
## Checklist

- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/search/src/components/ImportProductsToAlgolia/useSearchProvider.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const useSearchProvider = (algoliaConfiguration?: AlgoliaConfiguration) => {
+  const searchProvider = useMemo(() => {
+    if (!algoliaConfiguration?.appConfig?.appId || !algoliaConfiguration?.appConfig?.secretKey) {
+      return null;
+    }
+    return new AlgoliaSearchProvider({
+      appId: algoliaConfiguration.appConfig.appId,
+      apiKey: algoliaConfiguration.appConfig.secretKey,
+      indexNamePrefix: algoliaConfiguration.appConfig.indexNamePrefix,
+      enabledKeys: algoliaConfiguration.fieldsMapping.enabledAlgoliaFields,
```
Comment (lkostrowski) [inline apps/search/src/components/ImportProductsToAlgolia/useSearchProvider.ts:18]: you sure you want to execute this on frontend? You can call tRPC here and run algolia server side, so credentials are not exposed to the client

Comment (lkostrowski) [reply apps/search/src/components/ImportProductsToAlgolia/useSearchProvider.ts:18]: If you do this on the frontend, you should verify if app has proper permissions. We should ensure `MANAGE_APPS` is required to configure the app. If we agree on fetching credentials to the frontend, we should ensure credentials will not be retrieved by a staff user without MANAGE_APPS

Comment (andrzejewsky) [reply apps/search/src/components/ImportProductsToAlgolia/useSearchProvider.ts:18]: @lkostrowski i didn't touch that logic, it was only moved to the other file, it's even yours: https://github.com/saleor/apps/blob/main/apps/search/src/components/ImportProductsToAlgolia.tsx#L26

Comment (lkostrowski) [reply apps/search/src/components/ImportProductsToAlgolia/useSearchProvider.ts:18]: Fine. So maybe you should look into this later. When the app was created, it was not possible to open apps without MANAGE_APPS permission. Now, app can be accessed so probably we should verify that

File: apps/search/src/components/ImportProductsToAlgolia/index.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        </Box>
+      )}
+
+      {uploadState.type === "success" && (
+        <Box marginTop={5}>
+          <Text size={5} fontWeight="bold" as="p">
+            Import finished successfully!
+          </Text>
+        </Box>
+      )}
```
Comment (Cloud11PL) [inline apps/search/src/components/ImportProductsToAlgolia/index.tsx:59]: Question/suggestion: Is this visible enough? Afaik app-bridge allows apps to show toasts in dashboard.

**Review by Cloud11PL — APPROVED** [review]
Very nice, LGTM

**Pull request merged**